hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/WALCellCodec.java

```diff
@@ -256,20 +264,31 @@ public void write(Cell cell) throws IOException {
         }
       }
     }
+
+    private void writeCompressedValue(OutputStream out, Cell cell) throws IOException {
+      byte[] compressed = compression.getValueCompressor().compress(cell.getValueArray(),
+        cell.getValueOffset(), cell.getValueLength());
+      StreamUtils.writeRawVInt32(out, compressed.length);
+      out.write(compressed);
+    }
+
   }
 
   static class CompressedKvDecoder extends BaseDecoder {
     private final CompressionContext compression;
+    private final boolean hasValueCompression;
+    private final boolean hasTagCompression;
     public CompressedKvDecoder(InputStream in, CompressionContext compression) {
       super(in);
       this.compression = compression;
+      this.hasValueCompression = compression.hasValueCompression();
+      this.hasTagCompression = compression.hasTagCompression();
     }
 
     @Override
     protected Cell parseCell() throws IOException {
```

Comment:
Are similar updates needed in `SecureWALCellCodec`? my IDE is telling me that these methods between those two classes have a fair bit of duplicate code.

Reply:
Ugh, yes, WTF.
Ok, will do

Reply:
No, compression is not supported in `SecureWALCellCodec`. 

This is documented. You can have either compression or encryption, but not both. I'm sure this limitation can be addressed but would be separate follow up work.